### PR TITLE
chore: add next cache

### PR DESCRIPTION
**Pull request by claycrews2002**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -9,12 +9,43 @@ jobs:
...
+          # if source files changed but packages did not, rebuild from a prior cache
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # If there is no cache-hit, install dependencies 
+      - name: Install Dependencies 
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        uses: cypress-io/github-action@v5.0.5 # use the explicit version number
+        with:
+          build: yarn build
```
Comment (cravend) [inline .github/workflows/tests.yml:32]: I think you have to build below not here

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -9,12 +9,43 @@ jobs:
...
+          SPOTIFY_CLIENT_ID: ${{ secrets.SPOTIFY_CLIENT_ID }}
+          SPOTIFY_CLIENT_SECRET: ${{ secrets.SPOTIFY_CLIENT_SECRET }}
+          CYPRESS_SPOTIFY_USER: ${{ secrets.CYPRESS_SPOTIFY_USER }}
+          CYPRESS_SPOTIFY_PW: ${{ secrets.CYPRESS_SPOTIFY_PW }}
+
+      # Run all Cypress tests
       - name: Run E2E tests 🧪
         uses: cypress-io/github-action@v5.0.5 # use the explicit version number
         with:
-          build: yarn build
```
Comment (cravend) [inline .github/workflows/tests.yml:17]: Because if the cache exists the dependencies aren't installed so the other command doesn't run

**Review by cravend — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -9,12 +9,42 @@ jobs:
...
-      # and run all Cypress tests
+      # Get path of cache 
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      
+      - uses: actions/cache@v3
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
```
Comment (cravend) [inline .github/workflows/tests.yml:20]:
```suggestion
          path: |
            ${{ steps.yarn-cache-dir-path.outputs.dir }}
            ${{ github.workspace }}/.next/cache
```

This works for caching the Yarn install directory (yay!), but we need to make it do the Next cache as well. Based on the docs, I think you just need to add `${{ github.workspace }}/.next/cache` here, but experiment and see what works.

**Review by cravend — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 3 above)
Comment (claycrews2002) [reply .github/workflows/tests.yml]: You're right, I forgot to add that line after implementing for yarn instead of npm

Comment (cravend) [reply .github/workflows/tests.yml]: No worries, the documentation is super unclear for Yarn (I think Vercel has an anti-yarn vendetta)

**Review by cravend — APPROVED** [review]
Looks great!

**Pull request merged**